sdk/metric/sdk.go

```diff
@@ -336,39 +346,106 @@ func DefaultErrorHandler(err error) {
 // Labels returns a LabelSet corresponding to the arguments.  Passed
 // labels are de-duplicated, with last-value-wins semantics.
 func (m *SDK) Labels(kvs ...core.KeyValue) api.LabelSet {
-	// Note: This computes a canonical encoding of the labels to
-	// use as a map key.  It happens to use the encoding used by
-	// statsd for labels, allowing an optimization for statsd
-	// batchers.  This could be made configurable in the
-	// constructor, to support the same optimization for different
-	// batchers.
-
 	// Check for empty set.
 	if len(kvs) == 0 {
 		return &m.empty
 	}
 
-	ls := &labels{
-		meter:  m,
-		sorted: kvs,
+	ls := &labels{ // allocation
+		meter: m,
+		slice: kvs,
 	}
 
-	// Sort and de-duplicate.
-	sort.Stable(&ls.sorted)
+	// Sort and de-duplicate.  Note: this use of `ls.slice` avoids
+	// an allocation by using the address-able field rather than
+	// `kvs`.  Labels retains a copy of this slice, i.e., the
+	// initial allocation at the varargs call site.
+	//
+	// Note that `ls.slice` continues to refer to this memory,
+	// even though a new array is allocated for `ls.ordered`.  It
+	// is possible for the `slice` to refer to the same memory,
+	// although in the reflection code path of `computeOrdered` it
+	// costs an allocation to yield a slice through
+	// `(reflect.Value).Interface()`.
+	//
+	// TODO: There is a possibility that the caller passes values
+	// without an allocation (e.g., `meter.Labels(kvs...)`), and
+	// that the user could later modify the slice, leading to
+	// incorrect results.  This is indeed a risk, one that should
+	// be quickly addressed via the following TODO.
+	//
+	// TODO: It would be better overall if the export.Labels interface
+	// did not expose a slice via `Ordered()`, if instead it exposed
+	// getter methods like `Len()` and `Order(i int)`.  Then we would
+	// just implement the interface using the `orderedLabels` array.
+	sort.Stable(&ls.slice)
+
 	oi := 1
-	for i := 1; i < len(ls.sorted); i++ {
-		if ls.sorted[i-1].Key == ls.sorted[i].Key {
-			ls.sorted[oi-1] = ls.sorted[i]
+	for i := 1; i < len(kvs); i++ {
+		if kvs[i-1].Key == kvs[i].Key {
+			kvs[oi-1].Value = kvs[i].Value
```

Comment:
Add a comment here that you do this because the last Value for a Key overwrites previous Value for the same Key.

Reply:
Done.